example/tests/test_openapi.py

```diff
@@ -0,0 +1,168 @@
+        create_request(path),
+        {'get': 'list'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_path_with_id_parameter(snapshot):
+    path = '/authors/{id}/'
+    method = 'GET'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'get': 'retrieve'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_post_request(snapshot):
+    method = 'POST'
+    path = '/authors/'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'post': 'create'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_patch_request(snapshot):
+    method = 'PATCH'
+    path = '/authors/{id}'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'patch': 'update'}
+    )
+    inspector = AutoSchema()
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+def test_delete_request(snapshot):
+    method = 'DELETE'
+    path = '/authors/{id}'
+
+    view = create_view_with_kw(
+        views.AuthorViewSet,
+        method,
+        create_request(path),
+        {'delete': 'delete'}
+    )
+    inspector = AutoSchema()
+    # DRF >=3.12 changes the capitalization of these method mappings which breaks the snapshot,
+    # so just override them to be consistent with >=3.12
+    inspector.method_mapping = {
+        'get': 'retrieve',
+        'post': 'create',
+        'put': 'update',
+        'patch': 'partialUpdate',
+        'delete': 'destroy',
+    }
+    inspector.view = view
+
+    operation = inspector.get_operation(path, method)
+    snapshot.assert_match(json.dumps(operation, indent=2, sort_keys=True))
+
+
+@override_settings(REST_FRAMEWORK={
+    'DEFAULT_SCHEMA_CLASS': 'rest_framework_json_api.schemas.openapi.AutoSchema'})
+def test_schema_construction():
+    """Construction of the top level dictionary."""
+    patterns = [
+        re_path('^authors/?$', views.AuthorViewSet.as_view({'get': 'list'})),
+    ]
+    generator = SchemaGenerator(patterns=patterns)
+
+    request = create_request('/')
+    schema = generator.get_schema(request=request)
+
+    assert 'openapi' in schema
+    assert 'info' in schema
+    assert 'paths' in schema
+    assert 'components' in schema
+
+
+class TestSchemaRelatedField(TestBase):
+    def test_schema_related_serializers(self):
```

Comment:
all tests above are pytest style. To be consistent better to convert this to pytest as well.

Reply:
Yeah that was superfluous. I wonder why I did that?